pkgs/by-name/ru/rustls-libssl/package.nix

```
@@ -0,0 +1,84 @@
+{
+  lib,
+  stdenv,
+  llvmPackages,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  openssl,
+}:
+
+let
+  version = "0.2.1";
+  target = stdenv.hostPlatform.rust.rustcTargetSpec;
+  libExt = stdenv.hostPlatform.extensions.sharedLibrary;
+in
+rustPlatform.buildRustPackage {
+  pname = "rustls-libssl";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "rustls";
+    repo = "rustls-openssl-compat";
+    rev = "v/${version}";
+    hash = "sha256-/QSFrkFVSRBmpXHc80dJFnYwvVYceAFnoCtmAGtnmqo=";
+  };
+
+  # NOTE: No longer necessary in the next release.
+  sourceRoot = "source/rustls-libssl";
+
+  cargoHash = "sha256-Yyrs2eN4QTGGD7A+VM1YkdsIRUh3laZac3rsJThjTXM=";
+
+  nativeBuildInputs = [
+    pkg-config # for openssl-sys
+    llvmPackages.lld # build.rs specifies LLD as linker
+  ];
+  buildInputs = [
+    openssl
+  ];
+
+  doCheck = false; # TODO: can't find libcrypto
```

```suggestion
  preCheck = ''
    # tests dlopen libcrypto.so.3
    export LD_LIBRARY_PATH=${lib.makeLibraryPath [ openssl ]}
  '';
```
If the tests are indeed supposed to open openssl's libcrypto.

---

Oh, great suggestion, that did the trick!